src/vs/workbench/contrib/tasks/browser/taskQuickPick.ts

```diff
@@ -74,9 +77,29 @@ export class TaskQuickPick extends Disposable {
 		return '';
 	}
 
+	public static getTaskLabelWithIcon(task: Task | ConfiguringTask): string {
+		return task.configurationProperties.icon ? `$(${task.configurationProperties.icon}) ${task._label}` : task.configurationProperties.color ? `$(${Codicon.tools.id}) ${task._label}` : `${task._label}`;
+	}
+
+	public static applyColorStyles(task: Task | ConfiguringTask, entry: TaskQuickPickEntryType | ITaskTwoLevelQuickPickEntry, themeService: IThemeService): void {
+		if (task.configurationProperties.color) {
+			const colorTheme = themeService.getColorTheme();
+			const styleElement = getColorStyleElement(colorTheme);
+			entry.iconClasses = [getColorClass(task.configurationProperties.color)];
+			document.body.appendChild(styleElement);
+		}
+	}
+
 	private _createTaskEntry(task: Task | ConfiguringTask, extraButtons: IQuickInputButton[] = []): ITaskTwoLevelQuickPickEntry {
+		const customizeIconButton = { iconClass: ThemeIcon.asClassName(Codicon.pencil), tooltip: nls.localize('setIconAndColor', "Choose color and icon") };
 		const entry: ITaskTwoLevelQuickPickEntry = { label: this._guessTaskLabel(task), description: this._taskService.getTaskDescription(task), task, detail: this._showDetail() ? task.configurationProperties.detail : undefined };
-		entry.buttons = [{ iconClass: ThemeIcon.asClassName(configureTaskIcon), tooltip: nls.localize('configureTask', "Configure Task") }, ...extraButtons];
+		entry.buttons = [];
+		if (CustomTask.is(task)) {
```

Comment:
Why only have this for `CustomTask`? A `ContributedTask` or `ConfiguringTask` can both be modified in tasks.json to have a color and icon set.

Reply:
it won't work atm without API changes for the `ContributedTask` or `ConfiguringTask`

Reply:
I would still expect it to work for tasks in tasks.json though. A `ConfiguringTask`, is just data from tasks.json, so if the icon is defined in the tasks.json, we should be able to show it here.

Reply:
This was followed up on in https://github.com/microsoft/vscode/pull/152106 where it's added in `TaskDTO.to`